File: HM/matlab/HM_lik.m
function NegLL = HM_lik(a, ro, alpha, beta,w)

% modify =============================================
% v0 starting point (75% probability for a US following a CS+ in the acquisition phase)
v = [0.75 0.25];
% eta0: the associability of the preceding trial
eta = 1;
% modify end =========================================


T = length(a);

% loop over all trial
for t = 1:T

    % compute expectancy probabilities
    p = exp(beta*v) / sum(exp(beta*v));

    % compute expectancy probability for actual choice (0/1)
    expProb(t) = p(a(t));

    % update values
    PE = ro(t) - v(a(t));
    v(a(t)) = v(a(t)) + alpha * PE * eta(t);
    eta(t+1) = w * abs(PE) + (1-w) * eta(t);
    
    % compute expectancy value for actual choice (0/1)
    valueV(t) = v(a(t));

end

eta(1) = [];
% compute negative log-likelihood
NegLL = -sum(log(expProb));

File: HM/python/HM_lik.py
# %%
def HM_lik(response, data,ro,alpha,beta,w):
    import numpy as np

    eta = np.zeros(np.shape(data)[0]+1,dtype='float')    
    expProb = np.zeros_like(data,dtype='float')
    valueV = np.zeros_like(data,dtype='float')
    v = np.array([0,0.75])
    eta[0] = 1
    
    
    p1=np.zeros_like(data,dtype='float')
    p2=np.zeros_like(data,dtype='float')

    for t in range(len(data)):
        
        p = np.exp(beta*v) / np.sum(np.exp(beta*v))
        
        expProb[t] = p[data[t]]
        p1[t] = p[0]
        p2[t] = p[1]
        
        PE = ro[t]-v[data[t]]
        
        
        v[data[t]] = v[data[t]] + alpha * PE * eta[t]
        
        eta[t+1] = w * np.abs(PE) + (1-w) * eta[t]
        
        valueV[t] = v[data[t]]
        
    _,*eta = eta
    NegLL = -np.sum(response * np.log(expProb));
    eta = np.array(eta)
    return NegLL